CMakeLists.txt

```diff
@@ -115,7 +115,12 @@ set(MINIO_CPP_HEADERS
   include/miniocpp/utils.h
 )
 
+IF (WIN32)
 add_library(miniocpp STATIC ${MINIO_CPP_SOURCES} ${MINIO_CPP_HEADERS})
+ELSE ()
+add_library(miniocpp SHARED ${MINIO_CPP_SOURCES} ${MINIO_CPP_HEADERS})
```

Comment:
@harshavardhana You were talking about `CXXFLAGS` removal to use `minio-ccp` shared library. Can we add an example?

Reply:
actually that is not needed @balamurugana , I removed that requirement from the other project so miniocpp doesn't need that flag. 

CXX_ABI requirement is only needed for really old GCCs not necessary in this case.